src/figuresClasses.ts

```diff
@@ -1,11 +1,98 @@
-export interface Figure {}
+export interface Figure {
+  shape: string;
+  color: string;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: string = 'triangle';
+
+  color: string;
+
+  constructor(
+    color: string,
+    private a: number,
+    private b: number,
+    private c: number,
+  ) {
+    this.color = color;
+    this.validateSides();
+  }
+
+  private validateSides(): void {
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error('your error message');
+    }
+
+    if (
+      this.a + this.b <= this.c ||
+      this.a + this.c <= this.b ||
+      this.b + this.c <= this.a
+    ) {
+      throw new Error('your error message');
+    }
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+
+    return (
+      Math.floor(
+        Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c)) * 100,
+      ) / 100
+    );
+  }
+}
 
-export class Triangle implements Figure {}
+export class Circle implements Figure {
+  shape: string = 'circle';
 
-export class Circle implements Figure {}
+  color: string;
 
-export class Rectangle implements Figure {}
+  constructor(
+    color: string,
+    private radius: number,
+  ) {
+    this.color = color;
+    this.validateRadius();
+  }
+
+  private validateRadius(): void {
+    if (this.radius <= 0) {
+      throw new Error('your error message');
```

Comment:
The error message should be more descriptive. According to the checklist, provide a detailed description of the error, such as 'Radius must be greater than zero.'